modules/statistics/php/NDB_Form_statistics.class.inc

```diff
@@ -78,13 +88,312 @@ class NDB_Form_statistics extends NDB_Form
+                          count(DISTINCT(c.PSCID)) as val
+                      FROM candidate as c
+                          LEFT JOIN session s ON (s.CandID=c.CandID)
+                      WHERE c.CenterID <> '1'
+                          AND c.PSCID <> 'scanner'
+                          AND c.Active<>'Y'
+                          $ExtraSite_Criteria
+                          $ExtraProject_Criteria
+                      GROUP BY s.subprojectid",
+            array()
+        );
+
+        foreach($result as $row) {
+            $this->tpl_data['ps_inactive'][$row['rowid']] = $row['val'];
+            $this->tpl_data['ps_inactive']['total']      += $row['val'];
+        }
+
+        //MALE
+        $result = $DB->pselect(
+            "SELECT s.subprojectid as rowid,
+                          count(DISTINCT(c.PSCID)) as val
+                      FROM candidate as c
+                          LEFT JOIN session s ON (s.CandID=c.CandID)
+                      WHERE coalesce(s.active, 'Y')='Y'
+                          AND c.CenterID <> '1'
+                          AND c.PSCID <> 'scanner'
+                          AND c.Active='Y'
+                          AND c.Gender='Male'
+                          $ExtraSite_Criteria
+                          $ExtraProject_Criteria
+                      GROUP BY s.subprojectid",
+            array()
+        );
+
+        foreach($result as $row) {
+            $this->tpl_data['gender_male'][$row['rowid']] = $row['val'];
+            $this->tpl_data['gender_male']['total']      += $row['val'];
+        }
+        //FEMALE
+        $result = $DB->pselect(
+            "SELECT s.subprojectid as rowid,
+                          count(DISTINCT(c.PSCID)) as val
+                      FROM candidate as c
+                          LEFT JOIN session s ON (s.CandID=c.CandID)
+                      WHERE coalesce(s.active, 'Y')='Y'
+                          AND c.CenterID <> '1'
+                          AND c.PSCID <> 'scanner'
+                          AND c.Active='Y'
+                          AND c.Gender='Female'
+                          $ExtraSite_Criteria
+                          $ExtraProject_Criteria
+                      GROUP BY s.subprojectid",
+            array()
+        );
+
+        foreach($result as $row) {
+            $this->tpl_data['gender_female'][$row['rowid']] = $row['val'];
+            $this->tpl_data['gender_female']['total']      += $row['val'];
+        }
+
+        $result = $DB->pselect(
+            "SELECT DISTINCT
+                          s.subprojectid as rowid,
+                          AVG(DATEDIFF(c.Date_registered,c.DoB)) as age
+                      FROM candidate as c
+                          JOIN session s ON (s.CandID=c.CandID)
+                      WHERE coalesce(s.active, 'Y')='Y'
+                          AND c.CenterID <> '1'
+                          AND c.PSCID <> 'scanner'
+                          AND c.Active='Y'
```

Comment:
These queries should all have s.Active='Y' too, since it's joining the session table.